tools/src/main/scala/scala/scalanative/build/package.scala

```diff
@@ -0,0 +1,210 @@
+package scala.scalanative
+
+import java.nio.file.{Files, Path, Paths}
+import java.util.Arrays
+
+import scala.sys.process.Process
+
+import build.IO.RichPath
+import nir.Global
+
+package object build {
+
+  type LinkerPath = linker.ClassPath
+  val LinkerPath = linker.ClassPath
+
+  type LinkerReporter = linker.Reporter
+  val LinkerReporter = linker.Reporter
+
+  type LinkerResult = linker.Result
+  val LinkerResult = linker.Result
+
+  type OptimizerDriver = optimizer.Driver
+  val OptimizerDriver = optimizer.Driver
+
+  type OptimizerReporter = optimizer.Reporter
+  val OptimizerReporter = optimizer.Reporter
+
+  def build(nativelib: Path,
+            paths: Seq[Path],
+            entry: String,
+            target: Path,
```

Comment:
We already have `target` in Config that means something different (LLVM platform target). Lets pick a different name for this parameter to avoid potential confusion.